plugins/net_plugin/net_plugin.cpp

```diff
@@ -1081,6 +1096,25 @@ namespace eosio {
       });
    }
 
+   void connection::check_heartbeat( const tstamp &current_time )
+   {
+      if( protocol_version >= heartbeat_interval ) {
+         if( latest_msg_time > 0 &&  current_time > latest_msg_time + heartbeat_timeout ) {
+            if( !peer_addr.empty() ) {
+               fc_wlog(logger, "heartbeat timed out for peer address ${adr}", ("adr", peer_addr));
+               no_retry = benign_other;
+            } else {
+               fc_wlog(logger, "heartbeat timed out from ${p} ${ag}", ("p", last_handshake_recv.p2p_address)("ag", last_handshake_recv.agent));
+               no_retry = fatal_other;
```

Comment:
I don't see why you are setting `no_retry` to `fatal_other`. Why set it at all? For that matter why set it to `benign_other` above? That is set when sending a go_away message which we are not doing here.